docs/reverse-proxy.rst

```diff
@@ -64,14 +64,22 @@ If your proxy accepts both HTTP and HTTPS URLs, and you want your application
 to generate the appropriate url based on the incoming scheme, you'll want to
 pass waitress ``X-Forwarded-Proto``, however Waitress is also able to update
 the environment using ``X-Forwarded-Proto``, ``X-Forwarded-For``,
-``X-Forwarded-Host``, and ``X-Forwarded-Port``::
+``X-Forwarded-Host``, and ``X-Forwarded-Port``
+
+.. code-block:: nginx
 
    proxy_set_header X-Forwarded-Proto $scheme;
    proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
    proxy_set_header X-Forwarded-Host $host:$server_port;
    proxy_set_header X-Forwarded-Port $server_port;
 
-when using Apache, ``mod_proxy`` automatically forwards the following headers::
+Note that if your external proxy (for a pod or ingress) is serving a port that is different from your internal proxy (for a container), you may need to use ``$http_host`` to include the external port in ``request.route_url``. This is because the value of ``$server_port`` that is forwarded to your pyramid application will reference the value from the internal proxy, which is probably not what you want.
```

Comment:
This is `waitress`, there is no `request.route_url`. Waitress only knows about the WSGI environment. Documentation that references Pyramid or some other WSGI web framework is and appropriate here.

Also, the host header is controlled by the remote client. You generally do not want to use something that is controlled by the remote client, as a misconfiguration could leave you vulnerable to having URL's generated that are not valid for your application. It is a better idea to hardcode the host/port combination you send to Waitress at that point.